examples/step-98/step-98.cc

```diff
@@ -0,0 +1,974 @@
+     * 3. Computes the actual residual on the patch: `local_residual = local_rhs
+     * - A_patch * current_solution_on_patch`.
+     * 4. Applies the inverse patch operator (`inverse.vmult`) to the residual
+     *    to get a local correction: `local_correction = A_patch^{-1} *
+     * local_residual`.
+     * 5. Updates the global solution `dst` by distributing the negative of the
+     *    local correction.
+     *
+     * @param patch_storage The storage object containing patch information.
+     * @param dst The global solution vector (input/output).
+     * @param src The global right-hand side vector (input).
+     * @param patch_range The range of patch indices to process.
+     * @param do_forward Flag indicating the direction of iteration (forward or backward sweep).
+     */
+    void local_apply(const PatchStorageType                      &patch_storage,
+                     VectorType                                  &dst,
+                     const VectorType                            &src,
+                     const typename PatchStorageType::PatchRange &patch_range,
+                     const bool &do_forward) const override;
+
+    /**
+     * @brief Initialize the smoother.
+     *
+     * Initializes the base class and the local patch inverse operator.
+     *
+     * @param patch_storage Shared pointer to the patch storage object.
+     */
+    void initialize(std::shared_ptr<PatchStorageType> &patch_storage) override;
+
+    /**
+     * @brief Clear internal data.
+     *
+     * Clears the base class state and the local patch inverse.
+     */
+    void clear() override;
+
+  private:
+    /**
+     * @brief The tensor product matrix representing the patch operator.
+     * Provides the `apply_inverse` method.
+     */
+    InverseType patch_inverse_operator;
+  };
+
+  /**
+   * @brief Initialize implementation.
+   *
+   * Calls the base class initialize and initializes the `inverse` member if
+   * the patch storage contains any patches.
+   */
+  template <int dim, int fe_degree, typename number>
+  void LaplacePatchSmoother<dim, fe_degree, number>::initialize(
+    std::shared_ptr<PatchStorageType> &patch_storage)
+  {
+    // First, call the initialize function of the base class. This sets up
+    // the connection to the patch storage object.
+    BaseType::initialize(patch_storage);
+
+    // If there are no patches on the current processor, there is nothing to
+    // initialize, so we can exit early.
+    if (patch_storage->n_patches() == 0)
+      return;
+
+    // To build the local patch operator, we need information about the finite
+    // element and the geometry of the cells. We start by creating an
+    // FEEvaluation object, which gives us access to this information.
+    FEEval fe_eval(*patch_storage->get_matrix_free());
+
+    // The patch operator is constructed using a tensor product of 1D matrices.
+    // The deal.II FE_Q elements use a hierarchical basis, but the tensor
```

Comment:
I think this can be misunderstood. I suggest to be explicit
```suggestion
    // The deal.II FE_Q elements use a hierarchical ordering of the Lagrangian basis functions, but the tensor
```